Sources/SmithyJSON/Reader/Reader.swift

```diff
@@ -0,0 +1,230 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import protocol SmithyReadWrite.SmithyReader
+import enum SmithyReadWrite.Document
+import typealias SmithyReadWrite.ReadingClosure
+import enum SmithyReadWrite.ReaderError
+import enum SmithyTimestamps.TimestampFormat
+import struct SmithyTimestamps.TimestampFormatter
+import struct Foundation.Data
+import struct Foundation.Date
+import class Foundation.NSNull
+import class Foundation.NSNumber
+import func CoreFoundation.CFGetTypeID
+import func CoreFoundation.CFBooleanGetTypeID
+
+public final class Reader: SmithyReader {
```

Comment:
This type is the Reader used for deserializing JSON data to the Swift response types.